app/worker.go

```diff
@@ -0,0 +1,91 @@
+package app
+
+import log "github.com/Sirupsen/logrus"
+
+// nWorkers is how many workers this node has.
+const nWorkers = 10
+
+// Workers is a list of workers.
+var workers [nWorkers]*GenericWorker
+
+// Worker is an interface for the basic app worker tasks.
+type Worker interface {
+	HandleTransaction(work TransactionWork)
+	HandleBlock(work BlockWork)
+}
+
+// GenericWorker implements the basic worker.
+type GenericWorker struct {
+	ID int
+}
+
+// NewWorker returns a new TransactionWorker object.
+func NewWorker(id int) GenericWorker {
+	return GenericWorker{
+		ID: id,
+	}
+}
+
+// Start continually collects new work from existing work channels.
+func (w GenericWorker) Start() {
+	go func() {
+		for {
+			// Wait for work.
+			log.WithFields(log.Fields{
+				"id": w.ID,
+			}).Debug("Worker waiting for work.")
+			select {
+			case work := <-TransactionWorkQueue:
+				log.WithFields(log.Fields{
+					"id": w.ID,
+				}).Debug("Worker handling new transaction work.")
+				w.HandleTransaction(work)
+			case work := <-BlockWorkQueue:
+				log.WithFields(log.Fields{
+					"id": w.ID,
+				}).Debug("Worker handling new block work.")
+				w.HandleBlock(work)
+			case <-QuitChan:
+				log.WithFields(log.Fields{
+					"id": w.ID,
+				}).Debug("Worker quitting.")
+				return
+			}
+		}
+	}()
+}
+
+// NB: We're currently imposing a validation layer at the app level
+// using methods like SetUnsafe to get the transaction into the
+// pool, etc. Figure out a nicer way to do this, ie: where *should*
+// we validate?
```

Comment:
I thought `pool` would have a channel of incoming, unvalidated transactions that `app` could push onto. `pool` should do all *transaction* validation